ports/esp32/uart.c

```diff
@@ -27,21 +27,59 @@
  */
 
 #include <stdio.h>
-
+#include "uart.h"
 #include "driver/uart.h"
 #include "soc/uart_periph.h"
 
 #include "py/runtime.h"
 #include "py/mphal.h"
 
+#ifndef MICROPY_HW_UART_REPL_BAUD
+#define MICROPY_HW_UART_REPL_BAUD   115200
+#endif
+
 STATIC void uart_irq_handler(void *arg);
 
 void uart_init(void) {
+    uart_config_t uartcfg = {
+        .baud_rate = MICROPY_HW_UART_REPL_BAUD,
+        .data_bits = UART_DATA_8_BITS,
+        .parity = UART_PARITY_DISABLE,
+        .stop_bits = UART_STOP_BITS_1,
+        .flow_ctrl = UART_HW_FLOWCTRL_DISABLE,
+        .rx_flow_ctrl_thresh = 0
+    };
+    uart_param_config(UART_NUM_0, &uartcfg);
+
+    const uint32_t rxbuf = 129; // IDF requires > 128 min
```

Comment:
As pert stm32, maybe macro this 129 to `MICROPY_HW_UART_REPL_RXBUF`?

Reply:
This rxbuf is not used ( wasted ram :-( ) it needs their high level API ISR to work, but there's an assert to ensure it's defined.

Reply:
OK